### PR TITLE
feat(gatsby): Add aggregation resolvers

### PR DESCRIPTION
**Pull request by ascorbic**
Adds `min()`, `max()` and `sum()` resolvers to `allXXX` queries, similar to the current `group` and `distinct`.

Usage:

```graphql
{
  allShopifyProduct {
    maxPrice: max(field: variants___price)
    minPrice: min(field: variants___price)
    totalPrice: sum(field: variants___price)
  }
}
```

It supports numeric fields, but also attempts to cast non-numeric fields, and includes them if the value is not `NaN`.

### CODE REVIEW

File: packages/gatsby/src/schema/__tests__/queries.js [hunk 1]
```diff
@@ -1646,7 +1764,7 @@ describe(`Query schema`, () => {
     /**
      * queries are read from file and parsed with babel
      */
-    it.only(`escape sequences work when correctly escaped`, async () => {
```
Comment (vladar) [inline packages/gatsby/src/schema/__tests__/queries.js:1649]: 😬 Good catch!

**Review by vladar — APPROVED** [review]
Looks good to me!

File: packages/gatsby/src/schema/__tests__/queries.js (hunk 1 above)
Comment (ascorbic) [reply packages/gatsby/src/schema/__tests__/queries.js:1649]: I'm happy to say the other tests were all passing when I removed that!

**Comment by KyleAMathews** [comment]
This is great!

File: integration-tests/gatsby-source-wordpress/__tests__/__snapshots__/index.js.snap [hunk 2]
```diff
@@ -4,7 +4,7 @@ exports[`[gatsby-source-wordpress] Run tests on develop build Gatsby image proce
...
...\\"><span aria-hidden=\\"true\\" style=\\"width:100%;padding-bottom:66.66666666666667%\\"></span><img aria-hidden=\\"true\\" src=\\"data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAANABQDASIAAhEBAxEB/8QAGAAAAgMAAAAAAAAAAAAAAAAAAAQBAwX/xAAVAQEBAAAAAAAAAAAAAAAAAAABAP/aAAwDAQACEAMQAAAB1akIFoSC/8QAGxABAAEFAQAAAAAAAAAAAAAAAgABAxAREhT/2gAIAQEAAQUC6Ed4E+k4VZuf/8QAFhEAAwAAAAAAAAAAAAAAAAAAAAER/9oACAEDAQE/AYiI/8QAFhEAAwAAAAAAAAAAAAAAAAAAAAER/9oACAECAQE/Aays/8QAGBAAAgMAAAAAAAAAAAAAAAAAAAEgMTL/2gAIAQEABj8CsZlw/8QAGhABAQACAwAAAAAAAAAAAAAAAQARYSExUf/aAAgBAQABPyHTmwcodW0iwB4xwL//2gAMAwEAAgADAAAAEIjv/8QAFREBAQAAAAAAAAAAAAAAAAAAAGH/2gAIAQMBAT8QUn//xAAWEQEBAQAAAAAAAAAAAAAAAAAAEWH/2gAIAQIBAT8Qg0f/xAAcEAEAAgMAAwAAAAAAAAAAAAABACERQYFRkbH/2gAIAQEAAT8QxHzuWBMEIPZWYqGXIXKu3scAGw8x93uf/9k=\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;width:100%;height:100%;object-fit:cover;object-position:center;opacity:0;transition-delay:500ms\\"/><picture><source srcSet=\\"/static/72f3bac64c682555d9ad2ecceaffe51e/ab49a/1001-5616%C3%973744.jpg 75w,
```
Comment (ascorbic) [inline integration-tests/gatsby-source-wordpress/__tests__/__snapshots__/index.js.snap:7]: Ah dammit. Snapshot on an M1 with gloabl VIPS

Comment (ascorbic) [reply integration-tests/gatsby-source-wordpress/__tests__/__snapshots__/index.js.snap:7]:
```suggestion
<p><span class=\\"alignnone size-medium wp-image-7756 inline-gatsby-image-wrapper gatsby-image-wrapper\\" style=\\"position:relative;overflow:hidden;max-width:100%;width:300px\\" data-reactroot=\\"\\"><span aria-hidden=\\"true\\" style=\\"width:100%;padding-bottom:66.66666666666667%\\"></span><img aria-hidden=\\"true\\" src=\\"data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAANABQDASIAAhEBAxEB/8QAGAAAAgMAAAAAAAAAAAAAAAAAAAQBAwX/xAAVAQEBAAAAAAAAAAAAAAAAAAABAP/aAAwDAQACEAMQAAAB1akIFoSC/8QAGxABAAEFAQAAAAAAAAAAAAAAAgABAxAREhT/2gAIAQEAAQUC6Ed4E+k4VZuf/8QAFhEAAwAAAAAAAAAAAAAAAAAAAAER/9oACAEDAQE/AYiI/8QAFhEAAwAAAAAAAAAAAAAAAAAAAAER/9oACAECAQE/Aays/8QAGBAAAgMAAAAAAAAAAAAAAAAAAAEgMTL/2gAIAQEABj8CsZlw/8QAGhABAQACAwAAAAAAAAAAAAAAAQARYSExUf/aAAgBAQABPyHTmwcodW0iwB4xwL//2gAMAwEAAgADAAAAEIjv/8QAFREBAQAAAAAAAAAAAAAAAAAAAGH/2gAIAQMBAT8QUn//xAAWEQEBAQAAAAAAAAAAAAAAAAAAEWH/2gAIAQIBAT8Qg0f/xAAcEAEAAgMAAwAAAAAAAAAAAAABABEhQYFRkeH/2gAIAQEAAT8QqPvMgTBCD2VhWIW2GYwdvY4ANh5j7vc//9k=\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;width:100%;height:100%;object-fit:cover;object-position:center;opacity:0;transition-delay:500ms\\"/><picture><source srcSet=\\"/static/72f3bac64c682555d9ad2ecceaffe51e/ab49a/1001-5616%C3%973744.jpg 75w,
```

**Review by vladar — APPROVED** [review]

**Comment by nagiek** [comment]
Wanted to say thank you for this! It's a huge help!

**Comment by nagiek** [comment]
I did want to ask though if it were possible to add a “count” aggregation function? I don’t mind submitting a pull request.

**Pull request merged**